src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -1,116 +1,528 @@
-import React, { useState, useEffect, useCallback } from 'react';
-import { useHistory, useParams } from 'react-router-dom';
-import { Avatar, Grid, TextField, InputLabel } from '@material-ui/core';
+import React, { useEffect, useState } from 'react';
+import {
+  Avatar,
+  Grid,
+  TextField,
+  InputLabel,
+  Box,
+  IconButton,
+  Typography,
+} from '@material-ui/core';
+import { useSelector } from 'react-redux';
+import { RootStateType } from 'models/rootReducer';
+import { BasicButton } from 'components/Form';
+import { PhotoCamera } from '@material-ui/icons';
+import { Autocomplete } from '@material-ui/lab';
+import { uploadImageToImgur } from 'old/lib/utilities/Imgur/uploadImageToImgur';
+import { getStringFromFile } from 'old/lib/utilities/Imgur/getStringFromFile';
+import {
+  getCitiesByRegionId,
+  getRegionByCityId,
+} from 'old/lib/utilities/API/api';
 import { useStyles } from './styles/PersonalInfo.styles';
-import { IExpert } from '../../../old/lib/types';
-import { getCurrentUser } from '../../../old/lib/utilities/API/api';
-import { ERROR_404 } from '../../../old/lib/constants/routes';
+import { ICity, IRegion } from '../../../old/lib/types';
+import { ErrorField } from './ErrorField';
+import { regex } from './constants/regex';
+import { fields } from './constants/fields';
+import { INewAuthorValues } from './types';
 
 export const PersonalInfo: React.FC = () => {
   const classes = useStyles();
-  const { expertId } = useParams<{ expertId: string }>();
-  const [loadedExpert, setLoadedExpert] = useState<IExpert>();
-  const history = useHistory();
-
-  const fetchExpert = useCallback(async () => {
-    try {
-      const expertResponse = await getCurrentUser();
-      setLoadedExpert(expertResponse.data);
-    } catch (error) {
-      history.push(ERROR_404);
+  const regions = useSelector(
+    (state: RootStateType) => state.properties.regions,
+  );
+  const cities = useSelector((state: RootStateType) => state.properties.cities);
+  const [selectRegions, setSelectRegions] = useState<IRegion[] | null>(null);
+  const [regionForValue, setRegionForValue] = useState<IRegion | null>(null);
+  const [selectCities, setSelectCities] = useState<ICity[] | null>();
+  const [isCitySelect, setIsCitySelect] = useState<boolean>(false);
+  const [newAuthorValues, setNewAuthorValues] = useState<INewAuthorValues>({
+    avatar: '',
+    firstName: '',
+    lastName: '',
+    regionId: 0,
+    cityId: 0,
+    bio: '',
+    email: '',
+    /* eslint-disable-next-line */
+    socialNetwork: new Array(5).fill(null),
```

Comment:
I think that `[null, null, null, null, null]` is better(for readability and also for ESLint - why it complains btw?) to provide literally(since 5 does not come from external constant but is hard-coded)

Reply:
extra cool that we don't need to disable ESLint anymore.

just fyi, if we ever need to disable ESLint rule per-line, it's way better to disable specific rule(s):

```
// eslint-disable-next-line typescript-eslint/unsafe-assignment
```

this way we do many useful things:

1. keep catching _other_ issues
2. inform a reader what exactly was wrong - at that moment at least(since due to changes in other lines through time ESLint might stop complaining about this particular line so without a hint reader might be confused - "why this was disabled when there is no error? should I WORRY?!")